### DISCUSSION TITLE
Arguments not working

### DISCUSSION THREAD

**Discussion by Nuckerr** [Minestom/Minestom#1656] [Help] [answered]
Hey, hopfully this is the right channel to ask:
I have this code here: And it registers the player command (tho there are no suggestions) and the message argument just isn't there.
```kt
object KickCommand : Kommand({
    val playerTarget = ArgumentWord("player")
        .map { Manager.connection.getPlayer(it) ?: throw ArgumentSyntaxException("Invalid player", it, 1) }
        .suggest(SuggestionIgnoreOption.IGNORE_CASE) { Manager.connection.onlinePlayers.map { it.username } }
    val message = ArgumentType.StringArray("message")


    condition {
        sender.hasPermission("server.kick")
    }

    syntax(playerTarget, message) {
        if(player.uuid == (!playerTarget).uuid) {
            player.sendMessage("You can't kick your self, silly!".color(NamedTextColor.RED))
            return@syntax
        }
        (!playerTarget).kick("You have been kicked by an operator for ${(!message).joinToString(" ")}".color(NamedTextColor.RED))
        player.sendMessage("You have kicked ${(!playerTarget).username} for ${(!message).joinToString(" ")}")
    }.onlyPlayers()
```

**Comment by mworzala** [comment]
So the problem is that there is no message suggestion? Or something else?

**Reply by Nuckerr** [reply]
The first issue is there is no message argument. Just player argument

**Comment by Nuckerr** [comment]
bump

**Comment by DevBlokJochem** [comment, marked as answer]
Im currently working with the Command class instead of the Kommand class. 
So I dont know if you have to use command context, but for me using context is working fine.

This is an example how I did my tab completion & arguments: 
[KickCommand.txt](https://github.com/Minestom/Minestom/files/10400772/KickCommand.txt)

**Comment by MrBretze** [comment]
I dont have any problem, but i use a another Argument for getting the Player
Just replace BasicPermissionCommand by command and remove the useless thing to the super line

https://gist.github.com/MrBretze/7ffb1682b8c4361c6e58c6447ea3c3cc
